api/src/api.ts

```diff
@@ -112,6 +113,33 @@ export const plugin = {
       },
     });
 
+    server.route({
+      method: "GET",
+      path: "paper/{s2Id}",
+      handler: async (request) => {
+        const s2Id = request.params.s2Id;
+        let resp = null;
+        try {
+          resp = await s2Api.getPaperUncached(s2Id, config.s2.apiKey);
+        } catch (error) {
+          if (error.response?.status === 404) {
+            throw Boom.notFound(`not found: ${s2Id}`);
+          } else if (error.response?.status === 500) {
+            throw Boom.internal(`S2 API Error accessing paper ${s2Id}`);
+          }
+          throw Boom.internal(`Internal Error: ${error}`);
```

Comment:
Why not use a switch? It would be a lot more readable that way.
```
switch (error.response?.status) {
  case 404:
    throw Boom.notFound(`not found: ${s2Id}`);
  case 500:
    throw Boom.internal(`S2 API Error accessing paper ${s2Id}`);
  default:
    throw Boom.internal(`Internal Error: ${error}`);
}
```